workorder-job-repeater.lua

```diff
@@ -0,0 +1,296 @@
+-- reduces pauses between jobs of infinite work orders
+
+local function print_help()
+    print[====[
+
+workorder-job-repeater
+======================
+This script recreates workshop jobs upon completion in case
+they come from a work order with amount set to infinite. Usually
+a job for an infinite work order is only created once a day,
+resulting in dwarfs lazying around (this doesn't apply to non-infinite
+work orders). This script makes the dwarfes work harder.
+
+For best experience add following to your ``onMapLoad*.init``::
+
+    workorder-job-repeater start
+    
+**Usage**:
+
+``[enable | disable] workorder-job-repeater``
+
+``workorder-job-repeater start [<frequency>] | stop``
+
+``workorder-job-repeater --frequency <ticks> | --verbose | --help``
+
+:enable, start <frequency>: starts job monitoring, running every ``<frequency>`` ticks (default: 10).
+:disable, stop:             stops the script.
+
+-f, --frequency <ticks>     sets job monitoring frequency to ``<ticks>``
+-v, --verbose               toggles script's verbosity.
+-h, --help                  this help.
+
+]====]
+end
+
+local utils = require 'utils'
+local eventful = require 'plugins.eventful'
+
+local DEBUG = 1
+local INFO = 2
+local WARN = 3
+local ERROR = 4
+local OFF = 6
+
+local colors = {}
+colors[DEBUG] = COLOR_LIGHTMAGENTA
+colors[INFO] = COLOR_CYAN
+colors[WARN] = COLOR_YELLOW
+colors[ERROR] = COLOR_LIGHTRED
+
+-- globals persist across script runs
+verbosity = verbosity or INFO
+
+if not scriptname then
```

Comment:
Consider using `dfhack.current_script_name()` instead

Reply:
did that